### PR TITLE
Send stdout & stderr to Neptune channel

### PR DESCRIPTION
**Pull request by lukasz-walkiewicz**

### CODE REVIEW

File: neptune/internal/streams/stdstream_uploader.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    def flush(self):
+        pass
+
+
+class StdOutWithUpload(StdStreamUpload):
+
+    def __init__(self, experiment):
+        super(StdOutWithUpload, self).__init__(experiment, 'stdout', sys.__stdout__)
+        sys.stdout = self
```
Comment (dominik-jastrzebski) [inline neptune/internal/streams/stdstream_uploader.py]: I'm not sure if this trick will capture `stdout` and `stderr` output from external libraries that import `sys` on their own. Please check how Python import system works.

File: neptune/internal/streams/channel_writer.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    def __init__(self, experiment, channel_name):
+        self.time_started = time.time() * 1000
+        self._experiment = experiment
+        self._channel_name = channel_name
+        self._data = b''
+
+    def write(self, data):
+        self._data += data
+        lines = self.__SPLIT_PATTERN.split(self._data)
+        for i in range(0, len(lines) - 1):
```
[the remark this answers is not in the document]
Comment (lukasz-walkiewicz) [reply neptune/internal/streams/channel_writer.py]: This for skips last element of the list

File: neptune/internal/streams/stdstream_uploader.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        self._channel = experiment._get_channel(channel_name, 'text')
+        self._channel_writer = ChannelWriter(experiment, channel_name)
+        self._stream = stream
+
+    def write(self, data):
+        self._stream.write(data)
+        self._channel_writer.write(data)
+
+    def flush(self):
+        pass
```
[the remark this answers is not in the document]
Comment (lukasz-walkiewicz) [reply neptune/internal/streams/stdstream_uploader.py]: It is now

File: neptune/internal/streams/stdstream_uploader.py (hunk 1 above)
Comment (lukasz-walkiewicz) [reply neptune/internal/streams/stdstream_uploader.py]: There are a few cases when it won't work, e.g. when library uses stdout through variable assigned before this `stdout` swap but I still think that this solution should be fine in most cases.

**Pull request merged**